blaze/compute/numpy.py

```diff
@@ -55,6 +55,61 @@ def broadcast_ndarray(t, *data, **kwargs):
     compute_up.register(Broadcast, *([(np.ndarray, Number)] * i))(broadcast_ndarray)
 
 
+@dispatch(Repeat, np.ndarray)
+def compute_up(t, data, **kwargs):
+    if isinstance(t.lhs, Expr):
+        return np.char.multiply(data, t.rhs)
+    else:
+        return np.char.multiply(t.lhs, data)
+
+
+@dispatch(Repeat, np.ndarray, (np.ndarray, base))
+def compute_up(t, lhs, rhs, **kwargs):
+    return np.char.multiply(lhs, rhs)
+
+
+@dispatch(Repeat, base, np.ndarray)
+def compute_up(t, lhs, rhs, **kwargs):
+    return np.char.multiply(lhs, rhs)
+
+
+@dispatch(Concat, np.ndarray)
+def compute_up(t, data, **kwargs):
+    if isinstance(t.lhs, Expr):
+        return np.char.add(data, t.rhs)
+    else:
+        return np.char.add(t.lhs, data)
+
+
+@dispatch(Concat, np.ndarray, (np.ndarray, base))
+def compute_up(t, lhs, rhs, **kwargs):
+    return np.char.add(lhs, rhs)
+
+
+@dispatch(Concat, base, np.ndarray)
+def compute_up(t, lhs, rhs, **kwargs):
+    return np.char.add(lhs, rhs)
+
+
+@dispatch(Interp, np.ndarray)
+def compute_up(t, data, **kwargs):
+    if isinstance(t.lhs, Expr):
+        return np.char.mod(data, t.rhs)
+    else:
+        return np.char.mod(t.lhs, data)
+
+
+@dispatch(Interp, np.ndarray, (np.ndarray, base))
+def compute_up(t, lhs, rhs, **kwargs):
+    return np.char.mod(lhs, rhs)
+
+
+@dispatch(Interp, base, np.ndarray)
+def compute_up(t, lhs, rhs, **kwargs):
+    return np.char.mod(lhs, rhs)
```

Comment:
It won't work with the `@dispatch` decorator.  It should work if you use the `register` method on the Dispatcher itself.

https://github.com/mrocklin/multipledispatch/blob/master/multipledispatch/dispatcher.py#L74-L103